app/src/main/java/cloudcity/networking/models/ExtendedCellInfoModel.java

```diff
@@ -0,0 +1,224 @@
+package cloudcity.networking.models;
+
+import android.telephony.CellInfo;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Class combining {@link CellInfoModel} and {@link MeasurementsModel} into one
+ */
+public class ExtendedCellInfoModel {
+
+    private Integer earfcn;
+    private Integer pci;
+    private Long cellId;
+    private Integer eNodeBId;
+    @SerializedName("dummy_cell_info")
+    private Integer dummyCell;
+
+    // LTE
+    private Integer rsrp;
+    private Integer rsrq;
+    private Integer rssnr;
+
+    // 5G
+    private Integer csirsrp;
+    private Integer csirsrq;
+    private Integer csisinr;
+    private Integer ssrsrp;
+    private Integer ssrsrq;
+    private Integer sssinr;
+
+    @SerializedName("dummy_value")
+    private Integer dummyMeasurement;
+
+    @SerializedName("cell_type")
+    private int cellType;
+
+    public ExtendedCellInfoModel(
+            @NonNull MeasurementsModel measurementsModel,
+            @NonNull CellInfoModel cellInfoModel
+    ) {
+        super();
+
+        // Measurements stuff
+        setCsirsrp(measurementsModel.getCsirsrp());
+        setCsirsrq(measurementsModel.getCsirsrq());
+        setCsisinr(measurementsModel.getCsisinr());
+        setSsrsrp(measurementsModel.getSsrsrp());
+        setSsrsrq(measurementsModel.getSsrsrq());
+        setSssinr(measurementsModel.getSssinr());
+
+        setRsrp(measurementsModel.getRsrp());
+        setRsrq(measurementsModel.getRsrq());
+        setRssnr(measurementsModel.getRssnr());
+
+        setCellType(measurementsModel.getCellType());
+        setDummyMeasurement(measurementsModel.getDummy());
+
+        // Cell info stuff
+        setEarfcn(cellInfoModel.getEarfcn());
+        setPci(cellInfoModel.getPci());
+        setCellId(cellInfoModel.getCellId());
+        setDummyCell(cellInfoModel.getDummy());
+    }
```

Comment:
nah, lets not do this as I believe those are being checked before we get to initialize this class. fine suggestion though, but it makes no difference whether we crash due to IllegalArgumentException or NullPointerException really. and we'd like to know what we're crashing on instead of just vague "these things can't be null" message.

Reply:
done, resolving.